src/main/java/seedu/address/model/person/weblink/WebLink.java

```diff
@@ -0,0 +1,97 @@
+package seedu.address.model.person.weblink;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.person.weblink.WebLinkUtil.FACEBOOK_MATCH_STRING;
+import static seedu.address.model.person.weblink.WebLinkUtil.FACEBOOK_TAG;
+import static seedu.address.model.person.weblink.WebLinkUtil.INSTAGRAM_MATCH_STRING;
+import static seedu.address.model.person.weblink.WebLinkUtil.INSTAGRAM_TAG;
+import static seedu.address.model.person.weblink.WebLinkUtil.LINKEDIN_MATCH_STRING;
+import static seedu.address.model.person.weblink.WebLinkUtil.LINKEDIN_TAG;
+import static seedu.address.model.person.weblink.WebLinkUtil.TWITTER_MATCH_STRING;
+import static seedu.address.model.person.weblink.WebLinkUtil.TWITTER_TAG;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Represents a WebLink in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidWebLink(String)}
+ * only accept twitter, facebook, linkedin and instagram links
+ * WebLinkTag defines the category of this webLink.
+ */
+public class WebLink {
+
+    public static final String MESSAGE_WEB_LINK_CONSTRAINTS = "Hi, only valid facebook,"
+            + " instagram, linkedin or twitter links will be accepted:)";
+    public static final String WEB_LINK_VALIDATION_REGEX = "^^.*(|instagram.com|linkedin.com|twitter.com"
+            + "|facebook.com).*$";
+
+    public final String webLinkInput;
+    public final String webLinkTag;
+
+    /**
+     * Validates given web link.
+     *
+     * @throws IllegalValueException if the given webLink name string is invalid.
+     * TODO: better implementation in the future instead of hard code.
+     */
+    public WebLink(String name) throws IllegalValueException {
+        requireNonNull(name);
+        String trimmedWebLink = name.trim();
+        if (!isValidWebLink(trimmedWebLink)) {
+            throw new IllegalValueException(MESSAGE_WEB_LINK_CONSTRAINTS);
+        }
+        this.webLinkInput = trimmedWebLink;
+
+        if (webLinkInput.contains(FACEBOOK_MATCH_STRING)) {
```

Comment:
Use Switch Case instead? or simply use a regex to split them

Reply:
Yes i would think switch case is better.